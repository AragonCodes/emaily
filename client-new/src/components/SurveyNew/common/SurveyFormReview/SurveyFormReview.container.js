import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { surveyOperations } from '../../../../domain/survey';
import SurveyFormReviewComponent from './SurveyFormReview';

function mapStateToProps({ form }) {
	return { formValues: form.surveyForm.values };
}

const mapDispatchToProps = (dispatch) => {
	const submitSurvey = (values, history) => {
		dispatch(surveyOperations.submitSurvey(values, history));
	};

	return { submitSurvey };
};

const SurveyFormReviewContainer = connect(
	mapStateToProps,
	mapDispatchToProps
)(withRouter(SurveyFormReviewComponent));

export default SurveyFormReviewContainer;