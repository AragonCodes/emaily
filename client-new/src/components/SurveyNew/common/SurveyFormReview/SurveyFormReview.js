// SurveyFormReview shows users their form inputs for review
import React from 'react';
import formFields from '../formFields';

function SurveyFormReview({ onCancel, formValues, submitSurvey, history }) {
	function reviewFields() {
		return formFields.map(({ name, label }) => (
			<div key={name}>
				<label>{label}</label>
				<div>
					{formValues[name]}
				</div>
			</div>
		));
	}

	return (
		<div>
			<h5>Please confirm your entries</h5>
			{reviewFields()}
			<button
				className="yellow darken-3 white-text btn-flat"
				onClick={onCancel}
			>
				Back
      </button>
			<button
				onClick={() => submitSurvey(formValues, history)}
				className="green btn-flat right white-text"
			>
				Send Survey
        <i className="material-icons right">email</i>
			</button>
		</div>
	);
};

export default SurveyFormReview;
