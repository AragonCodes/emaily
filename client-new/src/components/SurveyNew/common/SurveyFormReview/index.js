import SurveyFormReviewContainer from './SurveyFormReview.container';

export default SurveyFormReviewContainer;