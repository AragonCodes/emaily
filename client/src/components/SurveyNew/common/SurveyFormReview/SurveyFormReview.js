// SurveyFormReview shows users their form inputs for review
import React from 'react';
import formFields from '../formFields';

function SurveyFormReview({ onCancel, formValues, submitSurvey, history }) {
	function reviewFields() {
		return formFields.map(({ name, label }) => (
			<div key={name}>
				<label>{label}</label>
				<div>
					{formValues[name]}
				</div>
			</div>
		));
	}

	return (
		<div>
			<h5>Please confirm your entries</h5>
			{reviewFields()}
			<div style={{ marginTop: '.5rem' }}>
				<button
					className="grey lighten-1 white-text btn-flat"
					onClick={onCancel}
				>
					Back
				</button>
				<button
					onClick={() => submitSurvey(formValues, history)}
					className="red accent-3 btn-flat right white-text"
				>
					Send Survey
        <i className="material-icons right">email</i>
				</button>
			</div>
		</div>
	);
};

export default SurveyFormReview;
