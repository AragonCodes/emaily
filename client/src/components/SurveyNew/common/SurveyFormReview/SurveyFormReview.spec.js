import React from 'react';
import { shallow } from 'enzyme';
import SurveyFormReview from './SurveyFormReview';

describe('SurveyFormReview', () => {
	it('should render correctly', () => {
		const wrapper = shallow(
			<SurveyFormReview
				onCancel={jest.fn()}
				submitSurvey={jest.fn()}
				formValues={{}}
				history={[]}
			/>
		);
		expect(wrapper.debug()).toMatchSnapshot();
	});
});