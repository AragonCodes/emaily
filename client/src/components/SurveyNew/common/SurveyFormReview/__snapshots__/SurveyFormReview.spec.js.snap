// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SurveyFormReview should render correctly 1`] = `
"<div>
  <h5>
    Please confirm your entries
  </h5>
  <div>
    <label>
      Campaign Title
    </label>
    <div />
  </div>
  <div>
    <label>
      Subject Line
    </label>
    <div />
  </div>
  <div>
    <label>
      Email Body
    </label>
    <div />
  </div>
  <div>
    <label>
      Recipient List
    </label>
    <div />
  </div>
  <div style={{...}}>
    <button className=\\"grey lighten-1 white-text btn-flat\\" onClick={[Function: mockConstructor]}>
      Back
    </button>
    <button onClick={[Function: onClick]} className=\\"red accent-3 btn-flat right white-text\\">
      Send Survey
      <i className=\\"material-icons right\\">
        email
      </i>
    </button>
  </div>
</div>"
`;
